src/main/java/org/terasology/hunger/HungerAuthoritySystem.java

```diff
@@ -41,18 +41,29 @@
 
 /**
  * @author UltimateBudgie <TheUltimateBudgie@gmail.com>
+ * The class monitoring player hunger levels, related events and commands.
  */
 
 @RegisterSystem(RegisterMode.AUTHORITY)
 public class HungerAuthoritySystem extends BaseComponentSystem implements UpdateSubscriberSystem {
+    /**
+     * The logger for debugging to the log files. =
+     */
     private static final Logger logger = LoggerFactory.getLogger(HungerAuthoritySystem.class);
 
+    /** Reference to the EntityManager, used for getting all entities who are affected by hunger. */
     @In
     private EntityManager entityManager;
 
+    /** Reference to the current time, used for calculating if the food of an entity has to be decreased.*/
     @In
     private Time time;
 
+    /**
+     * Get's the hunger of all entities available and checks for each one of those if their hunger is below the threshold of losing health.
```

Comment:
Don't go into too much detail on the implementation. Javadoc should be more "high-level", i.e., this method "updates entities with a HungerComponent and triggers DamageEvents if the hunger level drops below the threshold"